### PR TITLE
[5G: MVC] [SessionD] AMF Client Stub for SessionD Process

### PR DESCRIPTION
**Pull request by rupa455**
Signed-off-by: rupa455 <rupa.d@altencalsoftlabs.com>

## Summary

This file contains AMF Client Stub to send GRPC request to initiate SessionD process

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/session_manager/test/test_stubs/amf_client.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+namespace grpc {
+class Status;
+}  // namespace grpc
+namespace magma {
+namespace lte {
+
+class SetSMSessionContext;
+class SmContextVoid;
+
+}
```
Comment (sureshgorijavolu) [inline lte/gateway/c/session_manager/test/test_stubs/amf_client.h:20]: check indentation

File: lte/gateway/c/session_manager/test/test_stubs/amf_client.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ public:
+  AMFClient(AMFClient const&) = delete;
+  void operator=(AMFClient const&) = delete;
+
+ private:
+  AMFClient();
+  static AMFClient& get_instance();
+  std::unique_ptr<AmfPduSessionSmContext::Stub> stub_;
+  static const uint32_t RESPONSE_TIMEOUT = 10;  // seconds
+};
```
Comment (sureshgorijavolu) [inline lte/gateway/c/session_manager/test/test_stubs/amf_client.h:45]: check indentation

File: lte/gateway/c/session_manager/test/test_stubs/amf_client.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    PrintGrpcMessage(
+      static_cast<const google::protobuf::Message&>(request));
+    auto response_reader = client.stub_->AsyncSetAmfSessionContext(
+    local_response->get_context(), request, &client.queue_);
+    // Set the reader for the local response. This executes the `CreateSession`
+    // response using the response reader. When it is done, the callback stored in
+    // `local_response` will be called
+    local_response->set_response_reader(std::move(response_reader));
+}
```
Comment (sureshgorijavolu) [inline lte/gateway/c/session_manager/test/test_stubs/amf_client.cpp]: check indentation

**Comment by themarwhal** [comment]
Have you looked into creating the client test script in Python? It can be done much shorter. `magma/lte/gateway/python/scripts` for other client scripts.

**Review by themarwhal — CHANGES_REQUESTED** [review]
Please provide some documentation on how to use this AMF client stuf, although I think it might be simpler to do this in Python as I wrote before.

**Comment by rupa455** [comment]
Hi Marie, 

Yes marie, i am following python test scripts for creating the client.sorry for the late response. 

Thanks & Regards, 
Rupa.D

**Comment by rupa455** [comment]
> 
> 
> Please provide some documentation on how to use this AMF client stuf, although I think it might be simpler to do this in Python as I wrote before.


> 
> 
> Have you looked into creating the client test script in Python? It can be done much shorter. `magma/lte/gateway/python/scripts` for other client scripts.

Yes marie, i am following python test scripts for creating the client

**Comment by rupa455** [comment]
> 
> 
> Have you looked into creating the client test script in Python? It can be done much shorter. `magma/lte/gateway/python/scripts` for other client scripts.

Yes marie, i am following python test scripts for creating the client

**Comment by rupa455** [comment]
> 
> 
> Please provide some documentation on how to use this AMF client stuf, although I think it might be simpler to do this in Python as I wrote before.

Sure, we will provide the documentation.

**Pull request closed without merging**